### PR TITLE
Export the variables, write the VARS file and a new JSON file with details about the PR to be tested in all cases

### LINKED ISSUE #120: Export the variables, write the VARS file and a new JSON file with details about the PR to be tested in all cases
As discussed with @MalloZup at IRC:

We need to export the variables, write the VARS file and a write new JSON file with details about the PR to be on `--check` as well

Right now the VARS file is only written when an actual check was executed when a test was executed (no `--check`

This useful if at the real test we want to get information about the PR.

However the VARS file is just too simple and some tests will need more stuff such as the list of changes, and maybe even the changes themselves, or the author, time, etc.

So now a `.gitarro_pr.json` needs to be created created with basically all the information from the PR provided by octokit's `client.pull_request` and `client.pull_request_files`.

### PR DESCRIPTION
**Pull request by juliogonzalez**
## What does this PR do?

Fixes #120

## What issues does this PR fix or reference?

 - https://github.com/openSUSE/gitarro/issues/120

## Tests written? 

No, we do not have them even for the original .gitarro_vars file.

## Workflow:

https://github.com/openSUSE/gitarro/blob/master/doc/CONTRIBUTING.md

### CODE REVIEW

**Comment by juliogonzalez** [comment]
Two things about the failures:

- Locally I only had a complain about backend class being to big:
   ```
   (pr-info=)$ rubocop
   Warning: unrecognized cop Naming/UncommunicativeMethodParamName found in    .rubocop_todo.yml
   Warning: unrecognized cop Naming/UncommunicativeMethodParamName found in .rubocop.yml
   Inspecting 17 files
   ......C..........
   
   Offenses:
   
   lib/gitarro/backend.rb:158:1: C: Class has too many lines. [112/108]
   class Backend ...
   ^^^^^^^^^^^^^
   
   17 files inspected, 1 offense detected
   ```
   Not sure what to do here. I don't really think this can be solved with a refactor of a great part of the backend class.
- Why does the CI shows tests that are not related to changes of the PR and I can't see locally?

**Comment by MalloZup** [comment]
@juliogonzalez thx i will have a look tomorrow :+1:

**Comment by MalloZup** [comment]
@juliogonzalez  about CI, i think that the VERSION of rubocop change from TRAVIS and your local one.
This is might why the number of rubocop offenses vary and increase :)

**Comment by juliogonzalez** [comment]
@MalloZup could be but... but the only exception affecting this PR seems to be:
```
lib/gitarro/backend.rb:158:1: C: Metrics/ClassLength: Class has too many lines. [112/108]
class Backend ...
^^^^^^^^^^^^^
```
all other complains are unrelated to this PR and almost the same:
```
after them:
```
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/gitarro/backend.rb:320:5: C: Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.
    return false unless retriggered_by_comment?(pr.number, @context)
```

However I think you are right, maybe the rubocop got updates at Travis since the last time you merged something (6 months ago). Rubocop for Travis is now 0.59.2, while I have 0.59.

Anyway, the problems seems to be easy to fix, so ...

File: lib/gitarro/backend.rb [hunk 1]
```diff
@@ -99,8 +99,42 @@ def run_script
...
+       "GITARRO_PR_TITLE:  #{pr.title}\n" \
+       "GITARRO_PR_NUMBER: #{pr.number}\n" \
+       "GITARRO_PR_TARGET_REPO: #{@repo}\n")
+    end
+  end
+
+  def export_pr_data_to_file(pr)
+    pr = pr.to_hash
+    pr[:files] = []
+    @client.pull_request_files(@repo, pr[:number]).each do |ghfile|
```
Comment (MalloZup) [inline lib/gitarro/backend.rb]: pr[:files] = @client.pull_request_files(@repo, pr[:number]).map { |ghfile| (ghfile.to_hash }

Comment (MalloZup) [reply lib/gitarro/backend.rb]: i think this should work

Comment (MalloZup) [reply lib/gitarro/backend.rb]: https://mixandgo.com/learn/how-to-use-the-ruby-map-method

Comment (juliogonzalez) [reply lib/gitarro/backend.rb]: Mmm, I don't really understand this, and the syntax is invalid (there's a missing parentheses)

Can you explain?

**Pull request merged**